contracts/AckToEarn.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "hardhat/console.sol";
+
+contract AckToEarn is Ownable, ReentrancyGuard {
+    using Counters for Counters.Counter;
+
+    event NewBid(address indexed from, address indexed to, uint amount);
+    event ReclaimBid(address indexed bidder, uint amount);
+
+    Counters.Counter private bidIds;
+
+    struct Bid {
+        uint id;
+        address bidder;
+        address recipient;
+        uint amount;
+        uint timestamp;
+        string message;
+        bool claimed;
+        bool exists;
+    }
+
+    mapping(address => mapping(uint => Bid)) private bidderBids;
+    mapping(address => mapping(uint => Bid)) private recipientBids;
+
+    Bid[] public bids;
+
+    uint constant bidExpiryThreshold = 7 days;
+
+    /**
+    * @notice Send a message and bid amount to a recipient
+    */
+    function sendBid(string memory message, address recipient) external payable {
+        require(recipient != address(0), "Recipient cannot be the zero address");
+
+        // The recipient gets 90% of the bid amount
+        uint recipientAmount = (msg.value * 90) / 100;
+        uint newBidId = bidIds.current();
+        Bid memory bid = Bid({
+            id: newBidId,
+            bidder: msg.sender,
+            recipient: recipient,
+            amount: msg.value,
```

Comment:
should this be `recipientAmount`?

Reply:
This is the total amount of ether that was sent with the bid, including the 10% for the contract. I don't mind changing this to `recipientAmount` and deducting the 10% from this value. This would save us from continually having to do that calculation elsewhere.